### ISSUE TITLE
Proj0254 false positive: VersionPrefix '1.0.0' (and any other plain X.Y.Z value) reported as not SemVer-compliant

### ISSUE THREAD

**Issue by JohnVerheij** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#779]
`Directory.Build.props`:

```xml
<Project>
  <PropertyGroup>
    <VersionPrefix>1.0.0</VersionPrefix>
  </PropertyGroup>
</Project>
```

Any child project (verified on `net10.0` and `netstandard2.0`):

```
Directory.Build.props(N,M): warning Proj0254: VersionPrefix '1.0.0' does not comply with Semantic Versioning
```

`1.0.0` is the canonical valid SemVer 2.0 example. The same error fires for `0.23.0`, `2.5.7`,
and every other `X.Y.Z` value tested. The rule's documented "non-compliant" example
(`3.1.41596-beta0.03`) carries a `-prerelease` suffix that the rule legitimately wants out of
`VersionPrefix` and into `VersionSuffix`, but the rule's actual implementation also rejects
plain `MAJOR.MINOR.PATCH` values — i.e. essentially every non-prerelease version anyone would
ever set on a stable package.

Source-level check (`VersionPrefixShouldBeSemVerCompliant.cs`):

```csharp
private static bool NotSemantic(VersionPrefix version)
    => version.Value is not { PreRelease: null, BuildMetadata: null };
```

For `1.0.0`, `PreRelease` and `BuildMetadata` are both null and the underlying `SemVer.TryParse`
should succeed → `NotSemantic` should be `false`. Empirically it is `true`, so either
`SemVer.TryParse` is returning `null` for valid input or the property pattern is being inverted
somewhere upstream.

Sibling rule already reported by the maintainer: #773 (Proj0253 false positive on `2.3.2`).
Originating PR: #749 (shipped in 1.12.0).

Workaround: `<NoWarn>$(NoWarn);Proj0254</NoWarn>`.

**Comment by Corniel** [comment]
This is similar (but different) to #773 . Thanks for reporting this.

**Comment by Corniel** [comment]
@JohnVerheij It would be nice if you can review #783
